### ISSUE TITLE
Fee Bugs on EX 23-026408 (LP) - (Ready for UAT on PROD)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#12009]
### Fee duplication during portal payment on bill 7810939, payment failed to record on payment table.

- INC0282590 Created 4/12/23 & Victor notified
- CTM's analysis is in the comments [below](https://github.com/cityofaustin/atd-data-tech/issues/12009#issuecomment-1514945923). 

**What Happened:**
1. 3/6 System automated COS payment 7651065 for $410.80
2. The payment was voided by ATD user
3. 3/12 Permit applicant paid the bill via portal. 
    - ❌ The applicant was prompted to pay double the bill amount and did (see document [here](https://cityofaustin-my.sharepoint.com/:b:/g/personal/tracy_linder_austintexas_gov/EewCutAMdktLj-spUyOeXy0Be43W68unbcsM1vmO1cQAaQ?e=d20GjJ))
4. ❌ The system failed to record the payment on the fee payment table
5. ATD user canceled the bill

**Comment by TracyLinder** [comment]
### Duplicated Bug on Test
_Test ROWID 12880823_

1. (BO User) Apply for EX COS (Austin Water)
2. System applies COS payment
3. (BO User) Voids COS payment
4. (Portal User) Selects bill to be paid (bill 7555127, $410.80)
5. (City Base) Doubles the fee amount
6. (Portal User) Pays the fee ($821.60)
7. System did not apply the payment, bill shows unpaid

**Comment by TracyLinder** [comment]
### Analysis from CTM-Victor:

Regarding the automated COS payment procedure that is also mentioned in this ticket:

This procedure does not correctly update all of the relevant account* tables. At a quick glance, you can see that it updates the accountbill table to mark the bill as paid, but it does not add the payment amounts to the accountbillfee table.

The COS payment procedure needs to be analyzed and fixed so it will update all of the correct tables.

This payment procedure for COS is in COA_FEES.p_auto_payment

Starts at line 3870 through 3991

I'm not able to find any references to update the accountbillfee table in this procedure. This needs to be added.


![MicrosoftTeams-image (7)](https://user-images.githubusercontent.com/49930868/233126145-64a165c1-2f04-44fe-9032-c3b4ce2d49ef.png)

**Comment by TracyLinder** [comment]
FYI @Nadin-Nader this is ready to for our devs. Can you please assign this as a priority to Priya or Subha?

**Comment by Lpriya-LIT** [comment]
@Nadin-Nader Committed Changes and requested for Dev and Test deployment
SCTASK0306095

**Comment by Lpriya-LIT** [comment]
@TracyLinder @Nadin-Nader Ready for UAT on Test.
I followed the steps and verified the logic, now the fee is not doubled up. Please refer the screenshots

**Comment by Lpriya-LIT** [comment]
![image.png](https://images.zenhubusercontent.com/611eac1954217f265acc0499/cf737206-1eae-4344-badd-daa955a00537)

**Comment by Lpriya-LIT** [comment]
![image.png](https://images.zenhubusercontent.com/611eac1954217f265acc0499/4db6acfd-9ecb-4ae2-a159-c3dfb2455d78)

**Comment by TracyLinder** [comment]
### I will test this

**Comment by TracyLinder** [comment]
# UAT PASSED ✅ 
### Promote to PROD @Lpriya-LIT 

<img width="724" alt="Screenshot 2023-05-01 at 10 31 11 AM" src="https://user-images.githubusercontent.com/49930868/235478918-3242756f-fe60-4441-a32a-c1a4feddec89.png">
<img width="599" alt="Screenshot 2023-05-01 at 10 31 32 AM" src="https://user-images.githubusercontent.com/49930868/235478932-33cf5a47-d230-4594-8946-4d9b527e3f89.png">
<img width="625" alt="Screenshot 2023-05-01 at 10 32 54 AM" src="https://user-images.githubusercontent.com/49930868/235478952-9acee221-1f1d-4188-9da8-34e84056e85e.png">
<img width="607" alt="Screenshot 2023-05-01 at 10 33 19 AM" src="https://user-images.githubusercontent.com/49930868/235478960-eb63f0c1-5ad5-46b9-864d-df0fafdd51b5.png">

**Comment by Nadin-Nader** [comment]
Sent Kyle and Tracy a message saying this was ready for UAT on PROD.

**Comment by Nadin-Nader** [comment]
Sent Message to Kyle and ROW Team to please UAT and let us know if this is good to close or not now that it is on PROD. Waiting on Response

**Comment by Nadin-Nader** [comment]
This has been approved but cannot be tested due to the fact that it involves the payment of fees, but ROW is ok with closing and reopening if it is not working on PROD, Closing
